pkg/metricscollector/v1beta1/file-metricscollector/file-metricscollector_test.go

```diff
@@ -151,23 +235,39 @@ func TestCollectObservationLog(t *testing.T) {
 				},
 			},
 		},
+		"Missing objective metric in TEXT training logs": {
+			filePath:   filepath.Join(testTextDataPath, "missing-objective-metric.log"),
+			fileFormat: commonv1beta1.TextFormat,
+			metrics:    []string{"accuracy", "loss"},
+			expected: &v1beta1.ObservationLog{
+				MetricLogs: []*v1beta1.MetricLog{
+					{
+						TimeStamp: time.Time{}.UTC().Format(time.RFC3339),
+						Metric: &v1beta1.Metric{
+							Name:  "accuracy",
+							Value: consts.UnavailableMetricValue,
+						},
+					},
+				},
+			},
+		},
 	}
 
-	for _, test := range testCases {
-		t.Run(test.description, func(t *testing.T) {
+	for name, test := range testCases {
+		t.Run(name, func(t *testing.T) {
 			actual, err := CollectObservationLog(test.filePath, test.metrics, test.filters, test.fileFormat)
 			if (err != nil) != test.err {
 				t.Errorf("\nGOT: \n%v\nWANT: %v\n", err, test.err)
 			} else {
-				if !reflect.DeepEqual(actual, test.expected) {
-					t.Errorf("Expected %v\n got %v", test.expected, actual)
+				if diff := cmp.Diff(actual, test.expected); diff != "" {
+					t.Errorf("Expected %v\n got %v\ndiff: %v", test.expected, actual, diff)
```

Comment:
```suggestion
					t.Errorf("Unexpected parsed result: (-want,+got):\n%s", diff)
```

Only `diff` should be enough.

Reply:
I'll update it soon.